Failed tests (12666): TestChaos

The following test appears to have failed:

[#12666](https://circleci.com/gh/cockroachdb/cockroach/12666):

```
I0211 02:00:43.693975 27510 acceptance/chaos_test.go:160  client 2 shutting down
I0211 02:00:44.396214 27510 acceptance/chaos_test.go:160  client 0 shutting down
I0211 02:00:44.396305 27510 acceptance/chaos_test.go:193  round 2: restarting nodes [1 0 2]
W0211 02:00:44.396333 27510 acceptance/chaos_test.go:233  round 2: monkey sleeping while cluster recovers...
I0211 02:00:44.413322 27510 acceptance/chaos_test.go:307  699 120.7/sec
panic: test timed out after 5m0s

goroutine 221 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:703 +0x132
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0xcd8bd8, 0x1105c00, 0xe, 0xe, 0x728eadc4ff64c201)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc8204bcf08, 0xcd8440)
	/usr/local/go/src/testing/testing.go:494 +0x70
github.com/cockroachdb/cockroach/acceptance.TestMain(0xc8204bcf08)
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:140 +0x3e
main.main()
	github.com/cockroachdb/cockroach/acceptance/_test/_testmain.go:78 +0x113

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x1121fa0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0xbc

goroutine 6 [syscall, 4 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 19 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/acceptance.TestMain.func1()
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:131 +0xd8
created by github.com/cockroachdb/cockroach/acceptance.TestMain
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:139 +0x30

goroutine 136 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/acceptance.TestChaos.func4(0xc82004b2c0, 0xc820085e60, 0x3, 0x3)
	/go/src/github.com/cockroachdb/cockroach/acceptance/chaos_test.go:167 +0x3a
github.com/cockroachdb/cockroach/acceptance.TestChaos(0xc82015a7e0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/chaos_test.go:308 +0x19d0
testing.tRunner(0xc82015a7e0, 0x1105c30)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

goroutine 156 [select, 4 minutes]:
net/http.(*persistConn).readLoop(0xc82035eb00)
	/usr/local/go/src/net/http/transport.go:976 +0xac7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 25 [select, 4 minutes, locked to thread]:
runtime.gopark(0xcd9280, 0xc820025f28, 0xb9d648, 0x6, 0x431b18, 0x2)
	/usr/local/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc820025f28, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc820025f28)
	/usr/local/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 23 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fab413a68f8, 0x72, 0xc8200141f0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82018c920, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82018c920, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82018c8c0, 0xc820187000, 0x1000, 0x1000, 0x0, 0x7fab43c25050, 0xc8200141f0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200da530, 0xc820187000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820156370)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 24 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc820156370)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 196 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b2780)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 139 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fab413a65f8, 0x72, 0xc8200141f0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820556610, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820556610, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8205565b0, 0xc82059a000, 0x1000, 0x1000, 0x0, 0x7fab43c25050, 0xc8200141f0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82002c050, 0xc82059a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820156420)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 190 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b2500)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 213 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc82000b2c0)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 140 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc820156420)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 220 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc82000b360)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 158 [select, 4 minutes]:
github.com/docker/engine-api/client/transport/cancellable.Do.func3(0x7fab413b0988, 0xc820380400, 0xc8201ff400, 0xc820386d20)
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:78 +0xee
created by github.com/docker/engine-api/client/transport/cancellable.Do
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:84 +0x335

goroutine 152 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7fab413a66b8, 0x72, 0xc8200141f0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820385f70, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820385f70, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820385f10, 0xc820564000, 0x1000, 0x1000, 0x0, 0x7fab43c25050, 0xc8200141f0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82002c078, 0xc820564000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).monitor(0xc82018a000)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:459 +0x210
created by github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).Start
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:493 +0x3a7

goroutine 192 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b26e0)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 157 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc82035eb00)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 200 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b2820)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 202 [sleep]:
time.Sleep(0x3b9aca00)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/cockroachdb/cockroach/acceptance.TestChaos.func5(0xc82004b2c0, 0xc8204f5920, 0xc820150540, 0x3, 0xc820085e60, 0x3, 0x3, 0x7fab413b0b38, 0xc82018a000, 0xc82015a7e0, ...)
	/go/src/github.com/cockroachdb/cockroach/acceptance/chaos_test.go:235 +0x875
created by github.com/cockroachdb/cockroach/acceptance.TestChaos
	/go/src/github.com/cockroachdb/cockroach/acceptance/chaos_test.go:239 +0xf07
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/chaos_test.go
+++ b/acceptance/chaos_test.go
@@ -231,7 +231,7 @@ CREATE TABLE bank.accounts (
 
 			// Sleep until at least one client is writing successfully.
 			log.Warningf("round %d: monkey sleeping while cluster recovers...", curRound)
-			for !madeProgress() {
+			for !done() && !madeProgress() {
 				time.Sleep(time.Second)
 			}
 			log.Warningf("round %d: cluster recovered", curRound)

